### DISCUSSION TITLE
Login without pop-ups

### DISCUSSION THREAD

**Discussion by kornelski** [rust-lang/crates.io#4320] [Feature Requests]
I can't log in to crates.io from Firefox 92. The login pop-up opens, and then disappears, and literally nothing changes. No errors logged in the console. I can click "Log in" any number of times, and it will just keep opening the popup window, redirecting, and closing it.

It's very likely the fault of my browser or its extensions. However, the ephemeral pop-up window combined with JS-heavy page makes debugging of the problem very difficult. Top-level page doesn't log anything about GitHub auth process or its result, and the pop-up window is impossible to catch before it closes.

**Describe the solution you'd like**
Could you redirect to GitHub in the normal, top-level browsing context, rather than a pop-up window?

I suspect a more standard flow may also help with compatibility when multiple browsing context and or window openers aren't involved.

**Describe alternatives you've considered**
I can log in Chrome, but I'd prefer not to use it.

**Additional context**
![Screen Recording 2021-07-30 at 22 33 48](https://user-images.githubusercontent.com/72159/127713751-ca99441a-b939-43fe-bcb8-76468d95b598.gif)

**Comment by xkikeg** [comment]
+1 with Google Chrome Version 93.0.4577.82 (Official Build) (x86_64)

I could use Firefox Version 92.0 to sign-in, and then Google Chrome also could sign-in with the pop-up.

**Reply by Sfinx** [reply]
+1 with Chrome 98.0.4758.102 (Official Build) (64-bit), stock Ubuntu 20.04

**Reply by jackson211** [reply]
+1 with Chrome 100.0.4896.60 (Official Build) (x86_64) and macOS 12.3.1

**Reply by this-username-is-taken** [reply]
+1 with Chrome Version 103.0.5060.134 (Official Build) (arm64) and macOS 11.4 M1 chip

**Reply by greedy0110** [reply]
+1 with Chrome Version 105.0.5195.125 (Official Build) (x86_64) and macOS 12.5

**Comment by jmevel** [comment]
Just as a simple workaround for people passing by here.

If you want to login with your favorite browser still, you can check the HTTP request sent when clicking on the login button
It sends a `GET https://crates.io/api/private/session/begin`
Going to that URL gives us a tiny JSON

```json
{
    "state":"8Hj2QvfCxszwmPiumLBnwQ",
    "url":"https://github.com/login/oauth/authorize?response_type=code&client_id=9fe8110dfe185fe90b5c&state=8Hj2QvfCxszwmPiumLBnwQ&scope=read%3Aorg"
}
```

And copy pasting the URL from the JSON into your address bar brings you to a Github page where you can authorize crates.io to access your Github information if this is your first time.
Once it's done, go back to crates.io and click again on the login button. The website will detect you already authorized it to access your Github information and will create your account (or just log you in if you already have an account).

I know this isn't the solution to this issue but I thought it was worth sharing this workaround

**Reply by Turbo87** [reply]
interesting, thanks for sharing that. by figuring this out, did you get any clues why it did not work within the popup?

**Reply by jmevel** [reply]
Ok so after a bit of investigation it seems in my case the responsible for this issue was the Chrome extension [Ninja Cookie](https://chrome.google.com/webstore/detail/ninja-cookie/jifeafcpcjjgnlcnkffmeegehmnmkefl). After deactivating this extension the popup doesn't immediately close up and I'm able to login as I'm supposed to.

If anyone is interested, here are the logs with the working scenario (without any extension activated) and also the failing scenario (just Ninja Cookie activated). I took care of taking only the relevant parts of the logs starting from the click on the login button only.

[working.log](https://github.com/rust-lang/crates.io/files/8656525/working.log)
[failing.log](https://github.com/rust-lang/crates.io/files/8656526/failing.log)

**Comment by hhstore** [comment]
+1
